sequencer/src/block2.rs

```diff
@@ -614,4 +622,153 @@ mod test {
+        // - payload <4 bytes
+
+        setup_logging();
+        setup_backtrace();
+
+        let vid = test_vid_factory();
+        let num_test_cases = test_cases.len();
+        for (t, payload) in test_cases.into_iter().enumerate() {
+            let block = BlockPayload::from_bytes(payload);
+            tracing::info!(
+                "test payload {} of {} with {} txs",
+                t + 1,
+                num_test_cases,
+                block.len()
+            );
+
+            let disperse_data = vid.disperse(&block.payload).unwrap();
+
+            for index in block.iter() {
+                tracing::info!("tx index {}", index,);
+                let (tx, proof) = block.transaction_with_proof(&index).unwrap();
+                proof
+                    .verify(
+                        &tx,
+                        index,
+                        &vid,
+                        &disperse_data.commit,
+                        &disperse_data.common,
+                    )
+                    .unwrap()
+                    .unwrap();
+            }
+        }
+    }
+
+    fn tx_table(entries: &[usize]) -> Vec<u8> {
+        let mut tx_table = Vec::with_capacity(entries.len() + TxTableEntry::byte_len());
+        tx_table.extend(TxTableEntry::from_usize(entries.len()).to_bytes());
+        for entry in entries {
+            tx_table.extend(TxTableEntry::from_usize(*entry).to_bytes());
+        }
+        tx_table
+    }
+
+    fn random_tx_payloads_flat<R>(entries: &[usize], rng: &mut R) -> Vec<u8>
+    where
+        R: RngCore,
+    {
+        random_tx_payloads_flat_truncated(entries, 0, rng)
+    }
+    fn random_tx_payloads_flat_truncated<R>(
+        entries: &[usize],
+        max_tx_payloads_byte_len: usize,
+        rng: &mut R,
+    ) -> Vec<u8>
+    where
+        R: RngCore,
+    {
+        // largest entry dictates size of tx bodies
+        let mut result = vec![
+            0;
+            std::cmp::min(
+                *entries.iter().max().unwrap_or(&0),
+                max_tx_payloads_byte_len
+            )
+        ];
+        rng.fill_bytes(&mut result);
+        result
+    }
+
+    fn extract_tx_payloads(entries: &[usize], tx_payloads_flat: &[u8]) -> Vec<Vec<u8>> {
+        let mut result = Vec::with_capacity(entries.len());
+        let mut start = 0;
+        for end in entries {
+            let end = std::cmp::min(*end, tx_payloads_flat.len());
+            let tx_payload = if start >= end {
+                Vec::new()
+            } else {
+                tx_payloads_flat[start..end].to_vec()
+            };
+            start = end;
+            result.push(tx_payload);
+        }
+        assert_eq!(result.len(), entries.len());
+        result
+    }
+
+    fn random_payload<R>(entries: &[usize], rng: &mut R) -> Vec<u8>
+    where
+        R: RngCore,
+    {
+        random_payload_truncated(entries, 0, rng)
```

Comment:
Same here, why 0?

Reply:
Because I'm stupid and changed the function without changing its call sites. I'll fix that.

Reply:
fixed in https://github.com/EspressoSystems/espresso-sequencer/pull/846/commits/ab2a58b8e7bc9ac779722b3b28f3c108ac6b60b5